### PR TITLE
Vault secondary tier by reference

### PR DESCRIPTION
**Pull request by kkredit**
### :hammer_and_wrench: Description

No new features, just a couple minor improvements
- Recommend setting Vault performance secondary tiers via reference to primary
- Add 'computed: true' to Vault paths_filter
- Replace strings.EqualFold with ToLower in diff supression functions, since EqualFold was causing unexpected behavior

### :ship: Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-hcp/blob/main/CHANGELOG.md):

```release-note
NONE
```

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: internal/provider/resource_vault_cluster.go [hunk 1]
```diff
@@ -100,6 +100,7 @@ func resourceVaultCluster() *schema.Resource {
 					ValidateDiagFunc: validateVaultPathsFilter,
 				},
 				Optional: true,
+				Computed: true,
```
Comment (bcmdarroch) [inline internal/provider/resource_vault_cluster.go:103]: Let's drop this change for now!

File: internal/provider/resource_vault_cluster.go [hunk 2]
```diff
@@ -581,6 +581,8 @@ func setVaultClusterResourceData(d *schema.ResourceData, cluster *vaultmodels.Ha
 		} else {
 			d.Set("paths_filter", nil)
 		}
+	} else {
+		d.Set("paths_filter", nil)
```
Comment (himran92) [inline internal/provider/resource_vault_cluster.go:585]: just for my knowledge: do we need a new test for this or by default as not set for non-replicated cluster ->`paths_filter` was nil already

**Review by himran92 — APPROVED** [review]
lgtm, just a qs

**Review by bcmdarroch — APPROVED** [review]
Acceptance tests passed:
```
--- PASS: TestAccVaultCluster (1038.35s)
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/provider	1038.691s
```
